pkg/database/billing.go

```diff
@@ -0,0 +1,79 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+type billing struct {
+	c    cosmosdb.BillingDocumentClient
+	uuid string
+}
+
+// Billing is the database interface for BillingDocuments
+type Billing interface {
+	Create(context.Context, *api.BillingDocument) (*api.BillingDocument, error)
+	Get(context.Context, string) (*api.BillingDocument, error)
+	Patch(context.Context, string, func(*api.BillingDocument) error) (*api.BillingDocument, error)
```

Comment:
so never delete?

Reply:
Nope, billing service (running independently) will do it.

Reply:
Yes, the RP will not delete the record. The billing service will delete it after it has processed the record.